.github/workflows/test_coverage.yml

```
@@ -0,0 +1,21 @@
+name: Code Coverage
+
+on:
+  push:
+    branches: [ master, unstable ]
+  pull_request:
+    branches: [ master, unstable ]
+
+jobs:
+  test-coverage-upload:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Make this action like this:

```suggestion
      - uses: actions/checkout@v4
        with:
          ref: ${{ github.event.pull_request.head.sha }}
```

For more context on why, see this recently merged commit 60e1920f684e569c72bee0442099b6479c555b0a.

---

Oh wow, I didn't know GitHub Actions worked this way — thanks for the detailed insight!